moveit_ros/planning/moveit_cpp/include/moveit/moveit_cpp/planning_component.h

```diff
@@ -83,15 +115,54 @@ class PlanningComponent
     double max_velocity_scaling_factor;
     double max_acceleration_scaling_factor;
 
-    void load(const rclcpp::Node::SharedPtr& node)
+    template <typename T>
+    void declareOrGetParam(rclcpp::Node::SharedPtr const& node, std::string const& param_name, T& output_value,
+                           T default_value)
     {
+      // Try to get parameter or use default
+      if (!node->get_parameter_or(param_name, output_value, default_value))
+      {
+        RCLCPP_WARN(node->get_logger(),
+                    "Parameter \'%s\' not found in config use default value instead, check parameter type and "
+                    "namespace in YAML file",
+                    (param_name).c_str());
+      }
+    }
```

Comment:
How is this different from what was here before? Also, I don't see anything in this function that `declares` the parameter, only a call to `get_parameter_or` which depends on the parameter being declared already with some other mechanism.

Reply:
I added this to add a warning when a default value is used. I felt this might be helpful since it is very easy to mess up namespace and parameter declarations